### ISSUE TITLE
how to train a encoder that  encode a input to key and value? Could you detail the loss and other information?

### ISSUE THREAD

**Issue by biyaa** [seoungwugoh/opn-demo#1]
Hi,  I follow  your papers "Onion-Peel Networks for Deep Video Completion" and "A Unified Model for Semi-supervised and Interactive Video Object Segmentation using Space-time Memory Networks".  The papers all describe the encoder which output key and value, but don't konw how to propel encoder to output key(for addressing) and value (stores detailed information), what is the loss function? how to train encoder?
I appreciate it if you reply it .
thank you！

tim.credy in China

**Comment by seoungwugoh** [comment]
Hi, 
Keys and values are just feature maps, their roles are described according to the network structure. 
We do not give aditional losses for them, and end-to-end training for losses in the paper are enough to learn them.

**Comment by biyaa** [comment]
thank you
